src/main/java/it/gov/pagopa/pu/registry/mapper/pagopa/RegistryEventPagoPaDTO2PagoPaRegistryMapper.java

```diff
@@ -0,0 +1,80 @@
+package it.gov.pagopa.pu.registry.mapper.pagopa;
+
+import it.gov.pagopa.pu.registry.dto.RegistryEventPagoPaDTO;
+import it.gov.pagopa.pu.registry.enums.RegistryEventCategory;
+import it.gov.pagopa.pu.registry.enums.RegistryEventSubType;
+import it.gov.pagopa.pu.registry.enums.RegistryOutcome;
+import it.gov.pagopa.pu.registry.enums.RegistryPagopaEventType;
+import it.gov.pagopa.pu.registry.model.PagopaRegistry;
+import it.gov.pagopa.pu.registry.service.DataCipherService;
+import org.springframework.stereotype.Service;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.IntStream;
+
+@Service
+public class RegistryEventPagoPaDTO2PagoPaRegistryMapper {
+  private final DataCipherService dataCipherService;
+
+  public RegistryEventPagoPaDTO2PagoPaRegistryMapper(DataCipherService dataCipherService) {
+    this.dataCipherService = dataCipherService;
+  }
+
+  public List<PagopaRegistry> mapToPagoPaRegistry(RegistryEventPagoPaDTO dto) {
+    List<PagopaRegistry> result = new ArrayList<>();
+
+    String iuvRaw = dto.getIuv();
+    String navRaw = dto.getNav();
+
+    boolean iuvEmpty = (iuvRaw == null || iuvRaw.isBlank());
+    boolean navEmpty = (navRaw == null || navRaw.isBlank());
```

Comment:
let's use instead org.springframework.util.StringUtils#hasText(java.lang.String)
```suggestion
    boolean iuvEmpty = StringUtils.hasText(iuvRaw);
    boolean navEmpty = StringUtils.hasText(navRaw);
```